src/App/Fossa/ArchiveUploader.hs

```diff
@@ -78,47 +78,22 @@ archiveUploadSourceUnit baseDir apiOpts vendoredDeps = do
       updateArcName updateText arc = arc{archiveName = updateText <> "/" <> archiveName arc}
       archivesWithOrganization = updateArcName (T.pack $ show orgId) <$> archives
 
-  pure $ Just $ archivesToSourceUnit archivesWithOrganization
-
+  pure $ arcToLocator <$> archivesWithOrganization
 
 -- archiveNoUploadSourceUnit exists for when users run `fossa analyze -o` and do not upload their source units.
-archiveNoUploadSourceUnit :: [VendoredDependency] -> Maybe SourceUnit
-archiveNoUploadSourceUnit deps = Just . archivesToSourceUnit $ map forceVendoredToArchive deps
+archiveNoUploadSourceUnit :: [VendoredDependency] -> [Locator]
+archiveNoUploadSourceUnit deps = arcToLocator <$> map forceVendoredToArchive deps
```

Comment:
[nit]: This is essentially a `map f . map g a`, which is equivalent to `map (f . g) a`
```suggestion
archiveNoUploadSourceUnit deps = map (arcToLocator . forceVendoredToArchive) deps
```

Reply:
Why is this superior?

Reply:
`map f . map g` iterates over the list twice (in theory.  In practice, GHC might do some shenanigans to fix this, but we shouldn't rely on that).  `map (f .g)` iterates once, but does more transformation per iteration step.

Additionally, `map f $ map g a` is clearer than `f <$> map g a` even though they're identical, and `map (f . g) a` is an improvement on both.

Reply:
Ah, yea I see that now. Thanks for pointing that out. Didn't realize for some reason that I was iterating twice.